src/app/actions/majorRequirements.ts

```diff
@@ -85,12 +87,60 @@ export async function getMajorRequirements(
 export async function getAllRequirementLists(): Promise<RequirementList[]> {
   const requirementLists = await prisma.majorRequirement.findMany();
 
-  return requirementLists.map((reqList) => {
-    return parseRequirementList(reqList.requirementList as string);
+  const convertedRequirementLists: RequirementList[] = [];
+
+  requirementLists.forEach(async (reqList) => {
+    const r = await convertJSONToRequirementList(
+      reqList.requirementList as string,
+    );
+    convertedRequirementLists.push(r);
   });
+
+  return convertedRequirementLists;
+}
+
+// ==================================================================================
+// Helper functions start ===========================================================
+// ==================================================================================
+
+async function convertRequirementListToJSON(requirementList: RequirementList) {
+  const newReqList = await requirementListMapper((req) => {
+    return {
+      departmentCode: req.departmentCode,
+      number: req.number,
+    };
+  }, requirementList);
+
+  return JSON.stringify(newReqList);
 }
 
-// Helper functions
-function parseRequirementList(requirementList: string): RequirementList {
-  return JSON.parse(requirementList) as RequirementList;
+async function convertJSONToRequirementList(requirementListJSON: string) {
+  const requirementList = JSON.parse(requirementListJSON) as RequirementList;
+
+  return await requirementListMapper(async (req) => {
+    return await courseInfo({
+      departmentCode: req.departmentCode,
+      number: req.number,
+    });
+  }, requirementList);
+}
+
+// This function maps over a requirement list and applies a function to each class within the requirement list.
+// It uses (requirement: any) => any type to avoid making convoluted types to take just the departmentCode and number
+async function requirementListMapper(
+  f: (requirement: any) => any,
```

Comment:
Make this type more concrete. It should be async if you are awaiting it otherwise it can stay synchronous but then `requirementListMapper` doesn't have a need to be async. Also try to properly type the argument and return values.